src/main/java/foundation/stack/jdbc/NameGenerator.java

```diff
@@ -16,19 +16,12 @@
  * @author Ravi Chodavarapu (rchodava@gmail.com)
  */
 public class NameGenerator {
-    private static final String JDBC_DRIVER_PACKAGE = DockerDatabaseServerPerApplicationConnectionLookup.class.getPackage().getName();
 
     private static String getCallerClassName() {
         StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
         if (stackTrace != null && stackTrace.length > 0) {
-            for (int i = 1; i < stackTrace.length; i++) {
-                StackTraceElement element = stackTrace[i];
-                String className = element.getClassName();
-
-                if (!className.startsWith(JDBC_DRIVER_PACKAGE)) {
-                    return className;
-                }
-            }
+            int index = stackTrace.length == 1 ? 0 : stackTrace.length - 1;
```

Comment:
was the other one not working? neither is always going to be right but the older one is more likely to succeed. In a lot of cases, the bottom of the stack will contain JDK or library frames while starting from top till we encounter non jdbc will get an app frame.